src/Monitor/Monitor/ScheduledQueryRules/UpdateScheduledQueryRuleCommand.cs

```diff
@@ -0,0 +1,138 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Insights.OutputClasses;
+using Microsoft.Azure.Management.Monitor.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Monitor;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Management.Monitor.Management.Models;
+using Microsoft.WindowsAzure.Commands.Common.CustomAttributes;
+
+namespace Microsoft.Azure.Commands.Insights.ScheduledQueryRules
+{
+    /// <summary>
+    /// Updates a ScheduledQueryRule object
+    /// </summary>
+    [Cmdlet(VerbsData.Update, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ScheduledQueryRule",
```

Comment:
@shpimpal is there a reason we have both `Set-*` and `Update-*` cmdlets? If this API supports an `Update` call with an underlying `PATCH` operation, then we should just have a `New-*` cmdlet with an `Update-*` cmdlet. Otherwise, it should be a `New-*` cmdlet with a `Set-*` cmdlet.

Reply:
The API supports patch operation but only one field can be updated with the patch call, to update that parameter added `Update-*` cmdlet. So as per documentation, `New-*` cmdlet will create a new resource and `Set-*` can be used to update any of the fields of existing resource.

Reply:
@shpimpal would you mind explaining the `PATCH` API operation? You're saying that you can only update one property at a time in your `Update` SDK call?

Reply:
Yes. I meant that `PATCH` call to our SQR API accepts only one field of the resource that can be updated which is "Enabled". We cannot update other fields using `PATCH` call, we will have to use `Set-*` cmdlet to update any field of an existing resource which uses `PUT` semantics.

Reply:
@shpimpal in my opinion, since the only thing that you can change with this cmdlet is if the rule is enabled or not, it would be best to split it into two cmdlets: `Enable-AzScheduledQueryRule` and `Disable-AzScheduledQueryRule`

With this, users won't get confused by the fact there are two cmdlets with similar "updating a rule" experience, and then you have two cmdlets that allow the user to enable and disable a rule.